genesyscloud/consistency_checker/consistency_checker.go

```diff
@@ -240,27 +253,64 @@ func (c *ConsistencyCheck) CheckState() *retry.RetryError {
 				}
 
 				vv := v.New
-				if c.d.HasChange(k) {
+				if currentState.HasChange(k) {
 					if !compareValues(c.originalState[parts[0]], vv, slice1Index, slice2Index, key) {
-						return retry.RetryableError(&consistencyError{
+						err := retry.RetryableError(&consistencyError{
 							key:      k,
 							oldValue: c.originalState[k],
-							newValue: c.d.Get(k),
+							newValue: currentState.Get(k),
 						})
+
+						if _, exists := os.LookupEnv("BYPASS_CONSISTENCY_CHECKER"); c.checks >= c.maxStateChecks && exists {
+							c.writeConsistencyErrorToFile(currentState, err)
+							return nil
+						}
+
+						c.checks++
+						return err
 					}
 				}
 			} else {
-				if c.d.HasChange(k) {
-					return retry.RetryableError(&consistencyError{
+				if currentState.HasChange(k) {
+					err := retry.RetryableError(&consistencyError{
 						key:      k,
 						oldValue: c.originalState[k],
-						newValue: c.d.Get(k),
+						newValue: currentState.Get(k),
 					})
+
+					if _, exists := os.LookupEnv("BYPASS_CONSISTENCY_CHECKER"); c.checks >= c.maxStateChecks && exists {
+						c.writeConsistencyErrorToFile(currentState, err)
+						return nil
+					}
+
+					c.checks++
+					return err
 				}
 			}
 		}
 	}
 
-	DeleteConsistencyCheck(c.d.Id())
+	DeleteConsistencyCheck(currentState.Id())
 	return nil
 }
+
+func (c *ConsistencyCheck) writeConsistencyErrorToFile(d *schema.ResourceData, consistencyError *retry.RetryError) {
+	const filePath = "consistency-errors.log.json"
+	thing := consistencyErrorJson{
+		ResourceType:     c.resourceType,
+		ResourceId:       d.Id(),
+		GCloudObjectName: d.Get("name").(string),
```

Comment:
just need to update the var name from thing to something else. Also, i don't think it's a guarantee that name will be a field on the resource. Might be better to check if it exists and add it to this struct conditionally